pkg/profiler/profiler.go

```diff
@@ -80,7 +80,18 @@ type stackCountKey struct {
 
 type bpfMaps struct {
 	counts *bpf.BPFMap
-	traces *bpf.BPFMap
+	stacks *bpf.BPFMap
+}
+
+func (m bpfMaps) clean(stacks []int32, logger log.Logger) {
+	for _, stackId := range stacks {
```

Comment:
Is there a particular reason why we have changed read after delete logic? I'm just curious not to disagree.

Reply:
> Can we batch delete since we know the keys?

Was trying that out, with the idea of creating another PR for this, but it's not working on my machine yet (see the comment below)